src/clj/rems/db/entitlements.clj

```diff
@@ -36,6 +38,22 @@
                               :resource-ext-id resource-or-nil
                               :is-active? (not expired?)})))
 
+(defn- entitlement-to-permissions-api [{:keys [resid catappid start end mail userid approvedby]}]
+  (let [start-datetime (DateTime. start)]
+    (jwt/sign {:type     "ControlledAccessGrants"
+               :value    (str resid)
+               :source   "https://ga4gh.org/duri/no_org"
+               :by       (str approvedby)
+               :asserted (.getMillis start-datetime)} "secret"))) ;;TODO use key/real secret here
+
+(defn get-entitlements-for-permissions-api [user-or-nil resource-or-nil expired?]
+  {:ga4gh_visa_v1 (mapv entitlement-to-permissions-api
+                        (db/get-entitlements {:user            (if (has-roles? :handler :owner :organization-owner :reporter)
```

Comment:
you don't mention this behaviour in the swagger docs. would it be more explicit to do something like
```
(when-not (= user-or-nil (getx-user-id))
  (throw-unauthorized))